File: generate_diagram.py
import os
import re
from pathlib import Path
from mermaid.graph import Graph

def extract_mermaid_code(markdown_file):
    with open(markdown_file, 'r', encoding='utf-8') as f:
        content = f.read()
    mermaid_blocks = re.findall(r'```mermaid\n(.*?)\n```', content, re.DOTALL)
    return mermaid_blocks[0] if mermaid_blocks else None

def generate_svg_from_code(mermaid_code, output_file):
    graph = Graph(mermaid_code)
    svg = graph.to_mermaid()
    with open(output_file, 'w', encoding='utf-8') as f:
        f.write(svg)

if __name__ == '__main__':
    mermaid_code = extract_mermaid_code('README.md')
    if mermaid_code:
        os.makedirs('docs/images', exist_ok=True)
        generate_svg_from_code(mermaid_code, 'docs/images/architecture.svg')

File: generate_diagram.js
const fs = require('fs');
const path = require('path');
const mermaid = require('mermaid');

const inputPath = path.join(__dirname, 'README.md');
const outputPath = path.join(__dirname, 'docs', 'images', 'architecture.svg');

const markdownContent = fs.readFileSync(inputPath, 'utf-8');
const mermaidCodeMatch = markdownContent.match(/```mermaid\n([\s\S]+?)\n```/);

if (mermaidCodeMatch && mermaidCodeMatch[1]) {
  const mermaidCode = mermaidCodeMatch[1];
  const { svg } = mermaid.render('diagram', mermaidCode);

  fs.mkdirSync(path.dirname(outputPath), { recursive: true });
  fs.writeFileSync(outputPath, svg, 'utf-8');

  console.log('Diagram generated successfully!');
} else {
  console.error('No Mermaid code block found in README.md');
  process.exit(1);
}
